src/components/Input/Input.js

```diff
@@ -14,83 +14,81 @@ import {
+        <InputElementStyled
+          data-testid="input"
+          id={placeholder}
+          autoComplete="off"
+          value={value}
+          onChange={event => onChange(event.target.value)}
+          type={type}
+          onBlur={onBlur}
+          ref={reference}
+          aria-required={ariaRequired}
+          aria-invalid={ariaInvalid}
+          aria-describedby={`${placeholder}-desc`}
+          withIcon={icon}
+          floatingLabels={floatingLabels}
+        />
+        <LabelStyled
+          data-testid="input-label"
+          htmlFor={placeholder}
+          hasValue={value}
+          withIcon={icon}
         >
-          {/* {icon && <InputIconStyled>{icon.render()}</InputIconStyled>} */}
-          {required && <InputRequiredStyled>*</InputRequiredStyled>}
-          <InputElementStyled
-            id={placeholder}
-            autoComplete="off"
-            value={value}
-            onChange={event => onChange(event.target.value)}
-            type={type}
-            onBlur={onBlur}
-            ref={reference}
-            aria-required={ariaRequired}
-            aria-invalid={ariaInvalid}
-            aria-describedby={`${placeholder}-desc`}
-            withIcon={icon}
-            floatingLabels={floatingLabels}
-          />
-          <LabelStyled htmlFor={placeholder} hasValue={value} withIcon={icon}>
-            {placeholder}
-          </LabelStyled>
-          {showVisibilityIcon && (
-            <StyledButton
-              onClick={handleClickShowPassword}
-              tabIndex="0"
-              aria-label="toggle password visibility"
-              type="button"
-            >
-              {showPassword ? (
-                <StyledPasswordVisibility src={visibilityOff} alt="" />
-              ) : (
-                <StyledPasswordVisibility src={visibility} alt="" />
-              )}
-            </StyledButton>
-          )}
-        </InputElementWrapperStyled>
+          {placeholder}
+        </LabelStyled>
+        {showVisibilityIcon && (
+          <StyledButton
+            data-testid="input-visibility-icon"
+            onClick={handleClickShowPassword}
+            tabIndex="0"
+            aria-label="toggle password visibility"
+            type="button"
+          >
+            {showPassword ? (
+              <StyledPasswordVisibility src={visibilityOff} alt="" />
+            ) : (
+              <StyledPasswordVisibility src={visibility} alt="" />
+            )}
+          </StyledButton>
+        )}
+      </InputElementWrapperStyled>
 
-        <ErrorWrapper
-          passwordStrength={passwordStrength}
-          id={`${placeholder}-desc`}
-        >
-          {error}
-        </ErrorWrapper>
-      </InputComponentStyled>
-    );
-  }
-}
+      <ErrorWrapper
+        passwordStrength={passwordStrength}
+        id={`${placeholder}-desc`}
+      >
+        {error}
+      </ErrorWrapper>
+    </InputComponentStyled>
+  );
+};
 
 Input.propTypes = {
```

Comment:
I would put a time on the props definition here. Be as precise as we can witch prop validation here, eg. type should oneOf with defined available values